### ISSUE TITLE
Vibrato's wavy line extends to the next bar

### ISSUE THREAD

**Issue by WolfgangD** [CoderLine/alphaTab#471]
### Expected Results
Vibrato wavy line stops at the bar end or minimally after the bar ends. Comparable to Guitar Pros rendering.

### Observed Results
Currently, the wavy line stops a bit too late and extends into the next bar. This could potentially overlap with other notation and might hinder readability.

### Steps to Reproduce
1. Draw Notation Legend's bars 34 and 35

### Further details 
<img width="426" alt="Screenshot 2020-12-21 at 10 48 17" src="https://user-images.githubusercontent.com/1220328/102767227-82d77d00-437f-11eb-9a7d-012aa519021f.png">

### Your environment
**Version**
- 1.2 (develop, 72512ce)

**Flavor**
- JavaScript 

**System**
Mac OS Big Sur, Safari
options = {
                file: filename,
                core: {
                    engine: "svg"
                },
                player: {
                    soundFont: "FluidR3GM.sf2",
                    enableCursor: true,
                    enablePlayer: true,
                    enableUserInteraction: false,
                    vibrato: {
                        noteWideLength: 200,
                        noteWideAmplitude: 2,

                        noteSlightLength: 200,
                        noteSlightAmplitude: 1,

                        beatWideLength: 200,
                        beatWideAmplitude: 2,

                        beatSlightLength: 200,
                        beatSlightAmplitude: 1
                    },
                },
                notation: {
                    elements: {
                        ScoreTitle: false,
                        ScoreSubTitle: false,
                        ScoreArtist: false,
                        ScoreAlbum: false,
                        ScoreWords: false,
                        ScoreMusic: false,
                        ScoreWordsAndMusic: false,
                        ScoreCopyright: false,
                        GuitarTuning: false,
                        TrackNames: false,
                        EffectMarker: false,
                        EffectTempo: false
                    },
                    notationMode: 'songbook'
                }

**Comment by Danielku15** [comment]
What I assume here is: The vibrato line is drawn until the next note start beyond the bar. But in this case another full down and up line \/ is not possible due to the lack of space. In 1.1 I removed some duplicate whitespace that was added in front of the note so there used to be more space for lines making it more obvious. This behavior is in-line with the Guitar Pro drawing which I used as reference: 
![image](https://user-images.githubusercontent.com/674916/102899568-6749a080-446b-11eb-9b47-40d657cbf7ea.png)

But I agree that it looks a bit misleading in this case. We would have 2 possibilities here: 

1. I really add an option which would allow you to stop the wave line at the bar end instead of extending to the next bar. This setting would then likely affect various effect annotations behaving like this. e.g. like let-ring lines. 
2. I attempt drawing partial lines depending on the available space: 
![image](https://user-images.githubusercontent.com/674916/102900870-310d2080-446d-11eb-889a-1ddb0a0b374e.png)
